operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -656,11 +653,13 @@ private int numReplicasForConnectionDemand(double connectionDemand) {
     }
 
     private String getIngressControllerDomain(String ingressControllerName) {
-        return ingressControllerInformer.getList().stream()
-                .filter(ic -> ic.getMetadata().getName().equals(ingressControllerName))
-                .map(ic -> ROUTER_SUBDOMAIN + (ic.getStatus() != null ? ic.getStatus().getDomain() : ic.getSpec().getDomain()))
-                .findFirst()
-                .orElse("");
+        return Optional.ofNullable(informerManager.getLocalService(INGRESS_ROUTER_NAMESPACE, "router-" + ingressControllerName))
+                .map(Service::getStatus)
+                .map(ServiceStatus::getLoadBalancer)
+                .map(LoadBalancerStatus::getIngress)
+                .flatMap(l -> l.stream().findFirst())
+                .map(LoadBalancerIngress::getHostname)
+                .orElse(null);
```

Comment:
I don't know if it's really necessary, but previously this method would never have returned `null`.
```suggestion
                .filter(Objects::nonNull)
                .orElse("");
```

Reply:
> I don't know if it's really necessary, but previously this method would never have returned null.

We can return empty, but it doesn't seem necessary - go clients in particular can't differentiate between empty and missing past deserialization.

Reply:
> We can return empty, but it doesn't seem necessary

I'm fine with that, thanks